src/tests/test_randomtrackingdict.py

```diff
@@ -0,0 +1,48 @@
+"""
+Tests for RandomTrackingDict class
+"""
+import random
+import unittest
+
+from network import randomtrackingdict
+from time import time
+
+
+class TestRandomTrackingDict(unittest.TestCase):
+    """Main protocol test case"""
+
+    @staticmethod
+    def randString():
+        """helper function for tests, generates a random string"""
+        retval = b''
+        for _ in range(32):
+            retval += chr(random.randint(0, 255))
+        return retval
+
+    def test_check_randomtrackingdict(self):
+        """Check the logic of RandomTrackingDict class"""
+        a = []
+        k = randomtrackingdict.RandomTrackingDict()
+
+        a.append(time())
+        for i in range(50000):
+            k[self.randString()] = True
+        a.append(time())
+
+        while k:
+            retval = k.randomKeys(1000)
+            if not retval:
+                print('error getting random keys')
+
+            try:
+                k.randomKeys(100)
+                print('bad')
+            except KeyError:
+                pass
+            for i in retval:
+                del k[i]
+        a.append(time())
+
+        for x in range(len(a) - 1):
```

Comment:
This is debugging information. We should check if this is not too slow. Let's say all values of `a[x + 1] - a[x]` need to be less than `10.0`.